### ISSUE TITLE
getLocalIdent not working

### ISSUE THREAD

**Issue by VanishingDante** [webpack-contrib/css-loader#389]
I'm using css-loader@0.26.1 which is the latest version and Webpack@2.1.0-beta.25.

My css file loaders configuration is below
```
{
          test: /\.css$/,
          use: [
            { loader: 'style-loader' },
            {
              loader: 'css-loader',
              options: {
                module: true,
                // getLocalIdent: (loaderContext, localIdentName, localName, options) => `${loaderContext}-${localIdentName}-${localName}-${options}`,
                getLocalIdent: function (loaderContext, localIdentName, localName, options) {
                  console.log(arguments)
                  return 'whatever_random_class_name'
                },
                localIdentName: '[path]---[name]---[local]---[hash:base64:5]'
              }
            },
            { loader: 'postcss-loader' }
          ]
        },
```

no output in the console, the function is not even called

**Comment by nowaalex** [comment]
+1

**Comment by tonyhb** [comment]
+1

**Comment by fengyun2** [comment]
+1

**Comment by faceyspacey** [comment]
+1

**Comment by Lodin** [comment]
In accordance with results of my exploration there is no function `getLocalIdent` in [query](https://github.com/webpack/css-loader/blob/master/lib/loader.js#L15) object in `loader.js`. Looks like webpack2 converts everything in the `query` field to a JSON string before using query in loader, and that strips everything except JSON-compatible data. 

Hope it will help.

**Comment by soluml** [comment]
This is still an issue in webpack@2.2.0-rc.1

**Comment by soluml** [comment]
Tested this with latest Webpack (as of this writing, 2.2.0-rc.2) and I'm getting the same issues.  getLocalIdent is completely ignored.

**Comment by fengyun2** [comment]
css-loader@0.23.1 which is the latest version and Webpack@2.1.0-beta.28.
`localIdentName: [path]---[name]---[local]---[hash:base64:5]` 中的 `path` 没法识别

**Comment by huarse** [comment]
+ 1

**Comment by TanaseHagi** [comment]
```javascript
++ 1
```

**Comment by nshoes** [comment]
+1

**Comment by michael-ciniawsky** [comment]
Fixed by either installing webpack >= v2.2.1 or adding an `ident`  to the loader options

```js
{ 
     loader: 'css-loader`, 
     options: {
        ident: 'css',
        getLocalIdent: () => {}
        ...
     }
}
```
